src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -4221,7 +4222,17 @@ MapTaskList(MapMergeJob *mapMergeJob, List *filterTaskList)
 	}
 	else
 	{
-		partitionColumnName = ColumnName(partitionColumn, rangeTableList);
+		TargetEntry *targetEntry = tlist_member((Expr *) partitionColumn,
+												filterQuery->targetList);
+		if (targetEntry != NULL)
+		{
+			partitionColumnName = targetEntry->resname;
+		}
+
+		if (partitionColumnName == NULL)
```

Comment:
Maybe just add a comment that `targetEntry->resname` can be `NULL` so it is easier to understand why we are not doing an `else` here.